File: app/src/main/java/com/karla/learningverbs/view/verd_detail/adapters/ViewPagerAdapter.java
package com.karla.learningverbs.view.verd_detail.adapters;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.karla.learningverbs.R;
import com.karla.learningverbs.view.verd_detail.ViewPagerFragment;
import com.karla.learningverbs.model.ExampleVerb;
import com.karla.learningverbs.utils.LearningApplication;
import com.karla.learningverbs.utils.constants.Constants;

import java.util.ArrayList;

public class ViewPagerAdapter extends FragmentStateAdapter {

    private String[] arrayTitle = {LearningApplication.getInstance().getString(R.string.title_pager_present), LearningApplication.getInstance().getString(R.string.title_pager_past),LearningApplication.getInstance().getString(R.string.title_pager_future)};
    private ArrayList<ExampleVerb> exampleVerbs = new ArrayList<>();

    public ViewPagerAdapter(@NonNull FragmentActivity fm, ArrayList<ExampleVerb> exampleVerbs) {
        super(fm);
        this.exampleVerbs = exampleVerbs;
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        Fragment fragment = new ViewPagerFragment();
        Bundle args = new Bundle();
        args.putSerializable(Constants.ARG_OBJECT_VERB_EXAMPLE, exampleVerbs.get(position));
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public int getItemCount() {
        return arrayTitle.length;
    }

    public CharSequence getPageTitle(int position) {
        return arrayTitle[position];
    }
}

File: app/src/main/java/com/karla/learningverbs/kotlin/view/verbdetail/adapters/ViewPagerAdapter.kt
package com.karla.learningverbs.kotlin.view.verbdetail.adapters

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.karla.learningverbs.R
import com.karla.learningverbs.kotlin.utils.constants.Constants
import com.karla.learningverbs.kotlin.view.verbdetail.ViewPagerFragment
import com.karla.learningverbs.model.ExampleVerb
import com.karla.learningverbs.utils.LearningApplication2
import io.reactivex.annotations.NonNull

class ViewPagerAdapter(fragmentActivity: FragmentActivity, private var exampleVerbs :ArrayList<ExampleVerb>) : FragmentStateAdapter(fragmentActivity){

    private val arrayTitle = arrayOf(
        LearningApplication2.getInstance().getString(R.string.title_pager_present),
        LearningApplication2.getInstance().getString(R.string.title_pager_past),
        LearningApplication2.getInstance().getString(R.string.title_pager_future)
    )

    override fun getItemCount(): Int {
        return arrayTitle.size
    }

    @NonNull
    override fun createFragment(position: Int): Fragment {
        val fragment = ViewPagerFragment()
        val args = Bundle()
        args.putSerializable(Constants.ARG_OBJECT_VERB_EXAMPLE, exampleVerbs[position])
        fragment.arguments = args
        return fragment
    }


    fun getPageTitle(position: Int): CharSequence? {
        return arrayTitle[position]
    }


}